go-controller/pkg/node/gateway.go

```diff
@@ -229,6 +229,20 @@ func gatewayInitInternal(nodeName, gwIntf string, subnets []*net.IPNet, gwNextHo
 		return bridgeName, uplinkName, nil, nil, err
 	}
 
+	if config.OvnKubeNode.Mode == types.NodeModeSmartNIC {
+		// for smart-NIC we use the host IP/subnet provided by gwIPs
+		// as well as the host MAC address for the Gateway configuration
+		ips = gwIPs
+		hostRep, err := util.GetSmartNICHostInterface(bridgeName)
+		if err != nil {
+			return bridgeName, uplinkName, nil, nil, err
+		}
+		macAddress, err = util.GetSriovnetOps().GetRepresentorMacAddress(hostRep)
```

Comment:
As discussed in slack, this function can be changed to use devlink for querying port hardware address ( needs kernel >= 5.9.0).

Reply:
Indeed, `sriovnet` lib will be extended to support fetching the Port Representor MAC address via devlink as well.